tfjs-backend-webgpu/src/draw_test.ts

```diff
@@ -0,0 +1,194 @@
+    const startNumTensors = tf.memory().numTensors;
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    expect(tf.memory().numTensors).toEqual(startNumTensors);
+    expectArraysClose(
+        await readPixelsFromGPUCanvas(
+            canvas, height, width, unmultiplyAlpha, 'int32'),
+        data, 0.1);
+  });
+
+  it('draw image with 4 channels and float values', async () => {
+    // Premultiplied data.
+    const data =
+        [.1, .2, .3, .4, .5, .6, .7, .8, .09, .1, .11, .12, .13, .14, .15, .16];
+    const width = 2;
+    const height = 2;
+    const img = tf.tensor3d(data, [width, height, 4]);
+    const canvas = getCanvas();
+
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    const actualData =
+        await readPixelsFromGPUCanvas(canvas, height, width, unmultiplyAlpha);
+    expectArraysClose(actualData, data, 0.01);
+  });
+
+  it('draw image with 3 channels and int values', async () => {
+    const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    const width = 2;
+    const height = 2;
+    const img = tf.tensor3d(data, [width, height, 3], 'int32');
+    const canvas = getCanvas();
+
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    expectArraysEqual(
+        await readPixelsFromGPUCanvas(canvas, height, width, removeLastChannel),
+        data);
+  });
+
+  it('draw image with 3 channels and float values', async () => {
+    const data = [.1, .2, .3, .4, .5, .6, .7, .8, .9, .1, .11, .12];
+    const width = 2;
+    const height = 2;
+    const img = tf.tensor3d(data, [width, height, 3]);
+    const canvas = getCanvas();
+
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    const actualData =
+        await readPixelsFromGPUCanvas(canvas, height, width, removeLastChannel);
+    const expectedData = [25, 51, 76, 102, 128, 153, 178, 204, 229, 25, 28, 31];
+    // On macOs/M2, some difference is close to 1.
+    expectArraysClose(actualData, expectedData, 1);
+  });
+
+  it('draw 2D image in grayscale', async () => {
+    const data = [1, 2, 3, 4];
+    const width = 2;
+    const height = 2;
+    const img = tf.tensor2d(data, [width, height], 'int32');
+    const canvas = getCanvas();
+
+    tf.browser.draw(
+        // tslint:disable-next-line:no-any
+        img, canvas as any, {contextOptions: {contextType: env.name}});
+    const actualData = await readPixelsFromGPUCanvas(canvas, height, width);
+    const expectedData =
+        [1, 1, 1, 255, 2, 2, 2, 255, 3, 3, 3, 255, 4, 4, 4, 255];
+    expectArraysEqual(actualData, expectedData);
+  });
+
+  it('draw image with alpha=0.5', async () => {
+    const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
```

Comment:
What should happen if `data` is

```
[
  128, 128, 128, 128, 128, 128,
  255, 255, 255, 255, 255, 255,
]
```

Will the first line be darker than the second line when I put the canvas on screen?